contracts/OgvStaking.sol

```diff
@@ -216,9 +216,11 @@ contract OgvStaking is ERC20Votes {
     {
         require(duration >= minStakeDuration, "Staking: Too short");
         require(duration <= 1461 days, "Staking: Too long");
-        uint256 start = block.timestamp > epoch ? block.timestamp : epoch;
```

Comment:
epoch in deploy script: https://github.com/OriginProtocol/ousd-governance/blob/master/scripts/deploy.py#L6 is 86400 (seconds in a day). For that reason above will always default to `block.timestamp`

Reply:
On mainnet the epoch is correctly set to the start of the staking which is timestamp 1657584000 (Tuesday, July 12, 2022 12:00:00 AM UTC).

See https://etherscan.io/address/0x0c4576ca1c365868e162554af8e385dc3e7c66d9#readProxyContract.

It is NOT set to 86400

Reply:
Regarding the role of the epoch: 

    /// If staking before the start of staking (epoch), then the lockup start
    /// and end dates are shifted forward so that the lockup starts at the
    /// epoch.

https://github.com/OriginProtocol/ousd-governance/blob/f3b91ffbe7f54414418e42148420978235502d85/contracts/OgvStaking.sol#L93

Reply:
Got it thanks